grafana-plugin/src/components/EditModal/EditModal.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from 'react';
+
+import { css } from '@emotion/css';
+import { Alert, Button, Field, HorizontalGroup, Input, Modal, VerticalGroup, useStyles2 } from '@grafana/ui';
+import { ItemSelected } from 'core/types';
+
+export interface BaseEditModal {
+  isKeyEdit: boolean;
+  isOpen: boolean;
+  option: ItemSelected;
+  rowIndex: number;
+}
+
+export interface EditModalProps extends BaseEditModal {
+  valueField: string;
+  labelField: string;
+
+  onDismiss(): void;
+  onUpdateError(res: any): void;
+  onKeyUpdate(keyId: string, keyName: string, rowIndex: number): Promise<void>;
+  onValueUpdate(keyId: string, valueId: string, value: string, rowIndex: number): Promise<void>;
+}
+
+export const EditModal: React.FC<EditModalProps> = ({
+  isKeyEdit,
+  isOpen,
+  option,
+  onDismiss,
+  rowIndex,
+  onKeyUpdate,
+  onValueUpdate,
+  onUpdateError,
+
+  valueField: FieldId,
+  labelField: FieldName,
+}) => {
+  const styles = useStyles2(() => getStyles());
+
+  const [keyField, setKeyField] = useState<string>(option?.key[FieldName]);
+  const [valueField, setValueField] = useState<string>(option?.value[FieldName]);
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      title={<h2 className={styles.header}>{isKeyEdit ? 'Edit Key' : 'Edit Value'}</h2>}
+      onDismiss={onDismiss}
+    >
+      {isKeyEdit ? renderKeyEdit() : renderValueEdit()}
+    </Modal>
+  );
+
+  function getStyles() {
```

Comment:
1) It should be defined outside the component:
https://emotion.sh/docs/best-practices#consider-defining-styles-outside-your-components 
2) Just as I shared in PR made by Rares, I recommend that we use object syntax just as it is suggested in both official Emotion docs and Grafana style guide:
https://github.com/grafana/grafana/blob/main/contribute/style-guides/styling.md#usage
https://emotion.sh/docs/best-practices#use-typescript-and-object-styles
3) It should be either at the end of the file or in a separate file just as Rares did. We need to be consistent on this and decide on one or another way

Reply:
don't see at EditModal.tsx, core/types.ts and ServiceLablels please, they will be imported from `gops-labels`, sorry for confusion